collector/generic_collector.go

```diff
@@ -57,6 +62,63 @@ func (collector *GenericCollector) Name() string {
 
 //Returns collected metrics and the next collection time of the collector
 func (collector *GenericCollector) Collect() (time.Time, []v2.Metric, error) {
```

Comment:
There are a few things we do per collection that we should be able to do once at initialization:
- get minPollingFrequency
- Compile regexps